src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/ToToListConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.ConvertLinq.ConvertForEachToLinqQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery
+{
+    internal sealed class ToToListConverter : AbstractToMethodConverter
+    {
+        public ToToListConverter(
+            ForEachInfo<ForEachStatementSyntax, StatementSyntax> forEachInfo,
+            ExpressionSyntax selectExpression,
+            ExpressionSyntax modifyingExpression,
+            SyntaxTrivia[] trivia)
+            : base(forEachInfo, selectExpression, modifyingExpression, trivia) { }
+
+        protected override string MethodName => nameof(Enumerable.ToList);
+
+        /// Checks that the expression is "new List();"
+        /// Exclude "new List(a);" and new List() { 1, 2, 3}
+        protected override bool CanReplaceInitialization(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => expression is ObjectCreationExpressionSyntax objectCreationExpression && 
+               semanticModel.GetSymbolInfo(objectCreationExpression.Type, cancellationToken).Symbol is ITypeSymbol typeSymbol &&
+               CSharpConvertForEachToLinqQueryProvider.IsList(typeSymbol, semanticModel) &&
```

Comment:
? http://source.roslyn.io/#Microsoft.CodeAnalysis/WellKnownTypes.cs,57a154298e282143,references

Reply:
It seems to be inaccessible from here. Not sure if making it public is the right thing to do.

---
In reply to: [195850320](https://github.com/dotnet/roslyn/pull/27130#discussion_r195850320) [](ancestors = 195850320)